### PR TITLE
Make APIs internal when they should be - 1 - [Skip CI]

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
### Purpose of this PR
First pass of making APIs internal when they should be.

Note to reviewers:
Make sure that public APIs that I made internal should indeed not be public.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 namespace UnityEditor.Rendering.HighDefinition
 {
-    public abstract class BaseShaderPreprocessor
+    abstract class BaseShaderPreprocessor
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs]: Doesn't Users need this to write their own?
If not, Why do we need an abstract base class?

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/FullScreenCustomPassDrawer.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace UnityEditor.Rendering.HighDefinition
 	/// FullScreen custom pass drawer
 	/// </summary>
 	[CustomPassDrawerAttribute(typeof(FullScreenCustomPass))]
-    public class FullScreenCustomPassDrawer : CustomPassDrawer
+    class FullScreenCustomPassDrawer : CustomPassDrawer
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/FullScreenCustomPassDrawer.cs]: alignment issue?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightUtils.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@ namespace UnityEngine.Rendering.HighDefinition
     /// <summary>
     /// Light Utils contains function to convert light intensities between units
     /// </summary>
-    public class LightUtils
+    internal class LightUtils
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightUtils.cs]: Note: Internal is the default visibility for class directly within namespace (not nested). (The default is private if nested). So you can remove the keyword.

File: com.unity.render-pipelines.high-definition/Runtime/Debug/LightingDebug.cs [hunk 4]
```diff
@@ -33,7 +33,7 @@ public enum DebugLightFilterMode
         IndirectPlanarProbe = 1 << 8,
     }
 
-    public static class DebugLightHierarchyExtensions
+    internal static class DebugLightHierarchyExtensions
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/LightingDebug.cs]: Note: Internal is the default visibility for class directly within namespace (not nested). (The default is private if nested). So you can remove the keyword.

File: com.unity.render-pipelines.high-definition/Runtime/Debug/MaterialDebug.cs [hunk 5]
```diff
@@ -66,7 +66,7 @@ public enum MaterialSharedProperty
             Alpha,
         }
 
-        public class MaterialSharedPropertyMappingAttribute : Attribute
+        internal class MaterialSharedPropertyMappingAttribute : Attribute
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/MaterialDebug.cs]: Note: Internal is the default visibility for class directly within namespace (not nested). (The default is private if nested). So you can remove the keyword.

File: com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs (hunk 1 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs]: We have several different preprocessors for all our shaders, that's why there is a base abstract class. But we don't expect users to use the same basis a us.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/FullScreenCustomPassDrawer.cs (hunk 2 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/FullScreenCustomPassDrawer.cs]: Nope, just removed the public specifier.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightUtils.cs (hunk 3 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightUtils.cs]: Yeah I usually do it right :p
Must have been caught in a "internalization" spree of members and did it mechanically. Will fix other occurrences.

File: com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs (hunk 1 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/BaseShaderPreprocessor.cs]: hmm... currently we want, but as all this is not super well define let's not expose it

**Pull request merged**